### DISCUSSION TITLE
EN: CAN/Iso Shield for Fusion PCB is coming (including support for 2x Victron), DE: Fusion Platine bekommt CAN/Iso Aufsteckboard

### DISCUSSION THREAD

**Discussion by markusdd** [hoylabs/OpenDTU-OnBattery#955] [Show and tell]
English:
Hi, I know since I participated myself here that many people use the Fusion PCB.
We've got many requests for this, so we followed. There will be a 1x CAN 2x ISO Shield for the Fusion, which is super tiny, can be
put on the SPI Display header, AND is compatible with the PoE Shield as well.
The first batch has been ordered and should arrive just within a few weeks. The v1 Prototypes which just have 1x ISO will be available in limited numbers as well much sooner, so if you have just a single Victron these might suit you well.

Deutsch:
Hi, seitdem ich hier selbst Hilfe gesucht und gefunde habe, habe ich mitbekommen, dass viele hier auch die von mir designte Fusion-Platine nutzen und es große Nachfrage nach einer ISO/CAN erweiterung gibt. Diese wird es jetzt geben, die erste Charge ist im Zulauf.
Das Board ist super winzig, passt auf den SPI-Displayheader und ist sogar kompatibel, wenn ihr schon das PoE-Shield benutzt!
Wir haben eine begrenzte Stückzahl v1 Prototypen die nur 1x ISO haben, diese sind aber perfekt gut wenn ihr nur einen Victron nutzt. Diese werden etwas eher verfügbar sein.

Sneak peek:
<img width="374" alt="Bottom" src="https://github.com/helgeerbe/OpenDTU-OnBattery/assets/25175069/766f608f-7234-4ade-a64a-134a24218a24">
<img width="383" alt="Top" src="https://github.com/helgeerbe/OpenDTU-OnBattery/assets/25175069/a912eb90-d7a7-409b-bd5b-9393862c7e2a">

**Comment by markusdd** [comment]
Documentation now available.

Dokumentation ist jetzt verfügbar.

https://github.com/markusdd/OpenDTUFusionDocs/blob/main/CANIso.md

**Reply by Farquarl** [reply]
Please let me know when this is available.

**Reply by markusdd** [reply]
The v1 with support for 1 victron probably next week, the v2 as shown here probably end of this month.

**Reply by Farquarl** [reply]
Awesome, as i have two victrons i'll wait. Where will it be sold? 

Thanks a lot

**Reply by markusdd** [reply]
Yes same channel as the fusion.

**Reply by Farquarl** [reply]
Okay, reading your phrasing, it dawns on me that posting a specific shop here might be against the TOS. So I'll edit out the name.

**Reply by schlimmchen** [reply]
I have neither written nor read a TOS, but AFAIK you are welcome to mention the shop: [AllianceApps](https://shop.allianceapps.io/)

**Reply by markusdd** [reply]
as posted by Tim below, the v1s with 1 iso channel are now available in limited quantity.
We are selling them essentially 'at cost' for the community. The v2 with 2 serial ISO channels will come in a few weeks.
https://shop.allianceapps.io/products/opendtu-fusion-can-iso-shield-beta

**Reply by rudolfp1979** [reply]
v2 is available got one is working

**Comment by schlimmchen** [comment]
I love this and I am exited about this :blush: 

If you allow me to nag a little bit: It seems to be that the PCB will not be mounted very securely, only held in place by the pin header, correct? Even if it made for a weird PCB shape (and would increase cost), I would have considered an extension which reaches to the "upper left" mounting hole of the base PCB (and PoE shield), something like this:
![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/3578416/6c3877dc-ab60-4b07-b9d3-dca03f6fa11a)

The new CAN/Iso should could then be secured to a stud in the mounting hole.

**Reply by markusdd** [reply]
Yes. This has been discussed but we ended up not doing it for 5 reasons: 
a) such a shape would make assembly a pain
b) the shape would make panelling much more difficult 
c) this board is actually universal. The pinning is of course made for the fusion but in principle it can also be used by anyone with an existing setup by using just DuPont lines to connect it
d) the i2c connector is in the corner and if we sneak over there we might interfere with people's display mounting setups, also it limits compatibility with possible future shields
e) tension relief must be done on the case level anyway. None of the connectors are made for sustained physical load so you should be using a clamping pass-through anyway for your cables coming in.

Also, finally, that is the reason the board is so small: it actually is quite rigid sitting on top of the 8 pins there.

So overall this is a valid point but we decided to make that trade-off.

**Reply by schlimmchen** [reply]
Thanks for taking the time to reiterate the arguments! Very good points.

**Reply by Sorbat** [reply]
ich habe auf der Grundlage deiner Idee mal versucht etwas passendes in Fusion zu erstellen. Die Pins für RS458 und die ersten beiden Pins oben links werden dadurch zwar verdeckt, aber ich brauche die sowieso nicht und wenn man die braucht, könnte man ja noch eine Aussparung dafür machen. 
![fc1](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/72571258/bbe8347a-00fc-4a56-a5f7-e2b33ccb1cf7)
![fc1-2](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/72571258/552175a9-5046-4ee3-bd94-3d1d576ea4b1)
![fc2](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/72571258/46fc9829-5a49-43b7-8fac-0f849f7d418c)
also man kann es ja nach den eigenen wünschen anpassen https://www.thingiverse.com/thing:6660528

**Reply by markusdd** [reply]
sehr nett!

**Comment by Farquarl** [comment]
You want to drop any info on pricing?

**Reply by markusdd** [reply]
It won't be 50€ 😂

We'll have to pan out what our costs will be everything included (WEEE, CE, shipping, customs etc.) but it should be well below 20.
What makes it a little expensive are the analog devices isolators and the fact that it is 2-sided assembly. We made 125 I believe in the first batch  so that should spread the cost out per piece.

**Comment by cat24max** [comment]
"The v1 Prototypes which just have 1x ISO will be available in limited numbers as well much sooner, so if you have just a single Victron these might suit you well."

The v1 prototype is now available (15 pieces only): https://shop.allianceapps.io/products/opendtu-fusion-can-iso-shield-beta
Just keep in mind, this is a pre-production version.

**Reply by boennemann** [reply]
@cat24max Is there any way to debug whether the shield is working as intended?
I have a Pylontech/CAN as well as an MPPT/VE.Direct cable connected to the shield and I don't get a single log message – nothing. Both just don't work.
pin_mapping.json is updated as documented. VE.Direct set to "Normal Communication" in the app.

Problem is that this is an entirely new setup so I can't tell what's broken here, the shield, the cables I built, or something else. But also don't have any logs at all. I double checked the mapping of the VE.Direct cable, seems all fine. I can't believe that both cables I built are broken at the same time, but who knows.

**Reply by markusdd** [reply]
there is an option to turn on detailed logging.

Have you followed the guide from the docs repo and is the shield plugged in the right orientation?

My setup is 250km awway, is someone maybe able to take photos for @boennemann ?

**Reply by boennemann** [reply]
@markusdd Thanks for the quick reply!

Here is a photo of my setup for the moment.

<img width="724" alt="Screenshot 2024-05-28 at 19 42 04" src="https://github.com/helgeerbe/OpenDTU-OnBattery/assets/908178/87bca90c-e764-4fe9-9a0c-b47eda949a3a">

I followed the guide (loaded pin_mapping.json and switched the profile). Right orientation for all I can tell.

Verbose logging is enabled. Maybe I'm looking in the wrong location? I'm using Info > Console. I see literally nothing on there mentioning CAN/Battery/VE.Direct or what have you.
Connecting to Shelly/MQTT/Hoymiles all working.

I tested the VE.Direct cable with a Duspol. All individual cables have continuity on both ends. It's a crossover. Not sure what to test more.

**Reply by markusdd** [reply]
UART always has the potential to have TX/RX swapped and noo one has yet figured out a way to make it clear first try haha

I would honestly try to debug the CAN first as this is a little easier.

Do you have any means to debug this further in the sense of measuring if you see activity on the CANH/L lines?
(cheap oscilloscope maybe?)

what you can check on VE-direct with a multimeter is that you have 5V on the screw terminal (use the screws on top as contacts) between V+5V and VGND.

As for CAN: Is your cable a twisted pair for that? (I personally cut off one side of a Ethernet cable)

Also, very dumb question: is the battery turned on?

**Reply by boennemann** [reply]
> Do you have any means to debug this further in the sense of measuring if you see activity on the CANH/L lines?
> (cheap oscilloscope maybe?)

I fear no 😬 

> what you can check on VE-direct with a multimeter is that you have 5V on the screw terminal (use the screws on top as contacts) between V+5V and VGND.

Can confirm. Measuring 5V here.

> As for CAN: Is your cable a twisted pair for that? (I personally cut off one side of a Ethernet cable)

Bought a CAT5E ethernet cable, cut off one side, used 4 and 5

> Also, very dumb question: is the battery turned on?

There are no dumb questions :D Battery is on. I also still have some PV coming in. Inverter is also on and feeding into the house.

**Reply by markusdd** [reply]
ok, other basics to check: does the shield sit on the header without offset? (meaning that it is not shifted by a pin oor two to the right or left?)

Then: have you actually uploaded the pin mapping json? (I know this sounds ridiculous but I know why I'm asking, I did that myself before xD)

Also: Do you have the CAN and VE-Direct actually enabled (also in the dynamic power limiter?) Do you see the VE Direct and Pylontech sections on the homepage?

**Reply by boennemann** [reply]
Pin mapping uploaded and enabled. Numbers in the UI match the JSON. double checked. First had only one victron on, also tried two.

Shield firmly sits on top. I assume measuring the 5V confirms it's in the right location?

![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/908178/7852b189-45e8-42cf-bfb2-eb7597e8bfce)
![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/908178/cb318e32-89ea-49ed-9383-7775b87a8147)
![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/908178/a66f0fa0-8ad3-4324-b6f6-840b217d784a)


I didn't get to enabling the dynamic power limiter at all yet, because I'm not getting the necessary data.

Battery section is on the homepage with everything zero and an "no issues" next to it.

MPPT doesn't have its own section but MPPT power etc cards all showing 0.

**Reply by markusdd** [reply]
@schlimmchen is my memory correct that the limiter must be configured for anything to happen at all?

I think you might nt see anything because the DTU sees no need to query the data, could explain the empty logs as well.

Let me fire up my VPN and check the page of my remote setup.

**Reply by boennemann** [reply]
LOL that would be funniest outcome if true. And the worst part, just had to leave the house so I can't check for an hour-ish 🫣

**Reply by markusdd** [reply]
This is what you should see:
![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/25175069/c7c6c00f-bee6-46ee-ba72-cbf83a463774)
![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/25175069/72aa9527-9986-408c-a6a5-f4190fde79d8)
![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/25175069/c8498820-7008-4521-b3d5-ae267a909a76)
![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/25175069/e728ba0c-f0ea-4551-820a-21e8550d76f3)
![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/25175069/1a26d0d3-1c69-4ced-ac77-79fe89003972)

**Reply by schlimmchen** [reply]
The DPL may be off.

OpenDTU-OnBattery will poll the Pylontech battery if the battery interface is enabled and configured for Pylontech. If the pin mapping is off, you will see it on the USB interface when booting. Well, you could also save the battery settings, that will also trigger initialization and respective messages should appear in the web console.

Same for VE.Direct MPPT. If enabled in the web UI and if the pin mapping is correct, data will arrive and the VictronMpptController will process the data. If verbose logging is on, there will be a dump of the serial info.

I wired the Victron incorrectly as well the first time. The pin labelled "RX" on the addon board is supposed to be connected to the Victron RX pin. It is not labeled for the function at the ESP's side, but for the charge controller's pin.

**Reply by markusdd** [reply]
hahahaha yeah it's alwways the same with UART. RX and TX is literally THE WORST way to describe these lines, but unfortunately that is the world we live in now xD

I tried to deal with that in the pinout table but even in 10 years of professional digital IC Design I find there is no optimal way to communicate a UART pinout. It's ALWAYS wrong the first time xD

![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/25175069/ad97d014-180a-403e-8ede-ee74244738d8)

**Reply by boennemann** [reply]
I enabled the DPL, but as you wrote @schlimmchen, it didn't change anything.

> The pin labelled "RX" on the addon board is supposed to be connected to the Victron RX pin. It is not labeled for the function at the ESP's side, but for the charge controller's pin.

heh, well, that would explain it.
Will swap the cables now and report back.

With the VE.Direct cable I was more confident that I did it correctly, with the CAN cable, well not so much.

Let's see if VE.Direct starts working now. Next up CAN.

**Reply by markusdd** [reply]
😭 there is just no way to document UART the right way 😆

**Reply by boennemann** [reply]
What shall I say. It obviously immediately works.
I can feel the pain @markusdd. This really is undocumentable.
You could literally save all the writing down and just say try variant A, if it doesn't work try B. And I was SO sure that I did it right 🙈 


Thanks so much for the hand holding!

**Reply by markusdd** [reply]
hahahaha no worries.

I swear there is no way to do this right. You only get lucky or you don't xD

**Reply by boennemann** [reply]
And I got extra lucky, because I messed up the other cable, too. That lead me into questioning the shield in its entirety instead of the cables.

But it all works now. What a relief :)

**Reply by jaybanane** [reply]
> "The v1 Prototypes which just have 1x ISO will be available in limited numbers as well much sooner, so if you have just a single Victron these might suit you well."
> 
> 
> 
> The v1 prototype is now available (15 pieces only): https://shop.allianceapps.io/products/opendtu-fusion-can-iso-shield-beta
> 
> Just keep in mind, this is a pre-production version.

Hi, do you already know when the v2 version can be baught? I am sooo excited to get one and use it with 2 ve.direct channels.
Thanks a lot!

**Reply by cat24max** [reply]
> Hi, do you already know when the v2 version can be baught? I am sooo excited to get one and use it with 2 ve.direct channels. Thanks a lot!

Sorry, didn't get to it until now: https://shop.allianceapps.io/products/opendtu-fusion-can-iso-shield

**Comment by Manos1966** [comment]
> The pin labelled "RX" on the addon board is supposed to be connected to the Victron RX pin. It is not labeled for the function at the ESP's side, but for the charge controller's pin.

Manos66 laughing.... 🤣 🤣 🤣 🤣 🤣 
I could write books on the RX/TX story...

**Comment by Manos1966** [comment]
> Bought a CAT5E ethernet cable, cut off one side, used 4 and 5


![RJ45 wire colors](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/3a1d46b2-9fef-4e99-a0c0-97da68ec754e)
The cable colors of the RJ45 are  Blue and White/Blue

Is it your camera or, do I see in your picture Blue and **White/Green** ???

**Reply by boennemann** [reply]
I'm a tiny bit embarrassed now, but yes, you are right.
Thank you so much!

Will report back in a few min after rewiring.

**Reply by Manos1966** [reply]
Fingers crossed 🤞 

(AND NEVER ASK HOW I KNEW IT) 🙈

**Reply by boennemann** [reply]
Well … it works :) 🫶

And there I was wondering how both cables coming from the same pair would make so much more sense … well because it does make so much more sense apparently 🤦

**Reply by markusdd** [reply]
hahahaha nice.

glad it works.

Enjoy your setup!

**Reply by boennemann** [reply]
> Enjoy your setup!

I will! Currently enjoying all the million new Home Assistant entities and quite busy updating all the Dashboards.

I've gone through the Zendure Solarflow hell before and it's hard to describe how nerd-happy this all makes me. In a way I'm happy Zendure was so bad that it forced me to engage with all the new stuff that I learned and having a way better setup for it, too.

Super curious for the first days now.

**Reply by markusdd** [reply]
you will enjoy it a lot. in case you want to build a dashboard in Grafana I have open-sourced my battery and non-baterry version here on github: https://github.com/markusdd/solar_dashboard

![markus_solar_dashboard_battery](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/25175069/b34e9880-5635-4e35-abc3-a89264166613)

**Reply by Sorbat** [reply]
> The cable colors of the RJ45 are  Blue and White 

Hallo zusammen, ich hatte längere Zeit nach einem passenden Kabel gesucht. Ich wollte kein Netzwerkkabel nehmen und unbelegte Adern haben, sondern nur ein 2 poliges Kabel mit Mantel, dünn genug zum crimpen. Aber irgendwie waren alle Kabel, die ich getestet hatte, zu dick für den rj45.

Ich hatte dann gesehen, dass hier bei diesen Kabeln https://amzn.eu/d/ddvBGhc nur 4+5 mit blau/weiß belegt sind und sogar Stecker schon dran ist. Und glücklicherweise habe ich davon noch so ca 300 Stück im Keller :) Aber ich finde es eine ziemlich cleane und auch einfache Lösung.

**Comment by Manos1966** [comment]
I mention this here, just in case...

When it comes to the CANbus connection, you can use a standard RJ45 network cable (as long as you connect only the two cables (Blue & White/Blue) on the CAN-H & CAN-L side.

When it comes to the RS485 connection (we do not use in OpenDTU-OnBattery) it is IMPORTANT (if you use a RJ45 cable) to have ONLY Brown & White/Brown connected on BOTH sides of the cable! 
I have experienced Data corruption only because I connected a standard RJ45 network cable on the Pylontech side and using only two cables on the other side.

**Comment by Mickel4711** [comment]
Hello, sorry to chime in here. I want to use the CAN 1 ISO board. 
I only need CAN-H and CAN-L for this. 
Does this work without ground?

**Reply by Manos1966** [reply]
Pending confirmation from @markusdd  I would say yes:
GND1 and GND2 (as seen on the board picture) are aimed for the VE.Direct 1 and VE.Direct 2.
There is no GND anywhere else.

**Reply by markusdd** [reply]
yes. CAN Lines have no corresponding ground.
The ground pins are for the VE.Direct/other serial connections.

If CAN is your only use case you just need CANL/CANH.

**Comment by cat24max** [comment]
FYI, the final v2 version is now available to purchase: https://shop.allianceapps.io/products/opendtu-fusion-can-iso-shield
Order right now for shipping on Sunday :)

There is 2x v1 boards left, now available at a discount: https://shop.allianceapps.io/products/opendtu-fusion-can-iso-shield-beta

**Reply by schlimmchen** [reply]
> Order right now for shipping on Sunday :)

Enable checkout with a German address, then we can order :wink:

![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/3578416/693de150-e4f9-4499-abc8-4f52b30b215f)

**Reply by jaybanane** [reply]
Same here :(

**Reply by markusdd** [reply]
it's fixed now, just talked to Tim

**Reply by jaybanane** [reply]
Perfect, thanks!!!
